providers/gitlab/resources/gitlab.go

```diff
@@ -148,3 +148,31 @@ func (p *mqlGitlabProject) approvalSettings() (*mqlGitlabProjectApprovalSettings
 
 	return mqlApprovalSettings.(*mqlGitlabProjectApprovalSettings), nil
 }
+
+// To fetch protected branch settings
+func (p *mqlGitlabProject) protectedBranches() ([]interface{}, error) {
+	conn := p.MqlRuntime.Connection.(*connection.GitLabConnection)
+
+	projectID := int(p.Id.Data)
+	protectedBranches, _, err := conn.Client().ProtectedBranches.ListProtectedBranches(projectID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var mqlProtectedBranches []interface{}
+	for _, branch := range protectedBranches {
+		branchSettings := map[string]*llx.RawData{
+			"name":           llx.StringData(branch.Name),
+			"allowForcePush": llx.BoolData(branch.AllowForcePush),
+		}
+
+		mqlProtectedBranch, err := CreateResource(p.MqlRuntime, "gitlab.project.repository.protectedBranch", branchSettings)
+		if err != nil {
+			return nil, err
+		}
+
+		mqlProtectedBranches = append(mqlProtectedBranches, mqlProtectedBranch)
+	}
+
+	return mqlProtectedBranches, nil
+}
```

Comment:
I think once you add the id function here things will be good:
```
func (g *mqlGitlabProjectRepositoryProtectedBranch) id() (string, error) {
    return g.Name.Data, nil
}
```

Reply:
Thanks alot, It helped